### PR TITLE
disable migrations config

### PR DESCRIPTION
**Pull request by gary-alway**
### What does it do?

- Disable migrations given the `runMigrations` settings key
- Defaults to true if not present

### Why is it needed?

Model migrations mean that old migrations might not be compatible with the current schema which makes local development fail for fresh dbs.

### How to test it?

Passing `runMigrations` = `false` will prevent any migration scripts from running, by default migrations remain enabled.

### Related issue(s)/PR(s)

https://github.com/strapi/documentation/pull/1353

### CODE REVIEW

**Comment by gary-alway** [comment]
@alexandrebodin any news on this PR please ?

File: packages/core/database/lib/index.js [hunk 1]
```diff
@@ -33,7 +33,13 @@ class Database {
 
     this.schema = createSchemaProvider(this);
 
-    this.migrations = createMigrationsProvider(this);
+    if (config.connection.disableMigrations) {
+      this.migrations = {
+        shouldRun: () => Promise.resolve(false),
+      };
+    } else {
+      this.migrations = createMigrationsProvider(this);
```
Comment (alexandrebodin) [inline packages/core/database/lib/index.js:41]: this isn't in the right place in the config it should be part of settings instead. you will need to merge the incoming settings with the one on line 19 however. 

Also this condition should be handled in the migrations Provider shouldRUn function directly

**Comment by alexandrebodin** [comment]
I wasn't added as reviewer so didn't see this PR.

File: packages/core/database/lib/index.js (hunk 1 above)
Comment (gary-alway) [reply packages/core/database/lib/index.js]: thanks for the review @alexandrebodin 
I've updated as advised, please let me know

**Review by alexandrebodin — COMMENTED** [review]
Ok so general thought about the config name. we like to make the naming more declarative instead of enableX disableX, what do you think of `runMigrations: true|false` ?

File: packages/core/database/lib/index.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ class Database {
       connection: {},
       settings: {
         forceMigration: true,
```
Comment (alexandrebodin) [inline packages/core/database/lib/index.js:22]: You will need to add the default runMigrations: true here then :)

Comment (gary-alway) [reply packages/core/database/lib/index.js:22]: yes! good catch, thanks

**Comment by alexandrebodin** [comment]
Hey I'm seeing the CLA-sign not very happy probably because you used 2 different users to commit can you check that please :)

**Comment by pwizla** [comment]
Hi! Pierre, technical writer at Strapi here. Thank you for the contribution, @gary-alway!
This PR would require a small documentation update, thus creating a PR for Strapi's [documentation repo](https://github.com/strapi/documentation). If I understand correctly, we'd need to add a line to the table in [this section](https://docs.strapi.io/developer-docs/latest/setup-deployment-guides/configurations/required/databases.html#settings-configuration-object) to describe the `runMigration` parameter. 

Would you like to do it yourself, or would you rather have me handle it? 🙂

**Comment by gary-alway** [comment]
thanks @pwizla, I will have a go at update the docs

**Comment by gary-alway** [comment]
thanks @pwizla, @alexandrebodin 🚀 
keep up the great work on Strapi, its a pleasure to work with

**Pull request merged**